dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/CompanyDataControllerGetCompaniesEndpointTest.kt

```diff
@@ -0,0 +1,287 @@
+        val expectedDataMetaInfo = uploadTestDataSet(uploadInfo.actualStoredCompany.companyId)
+        val expectedStoredCompany = uploadInfo.actualStoredCompany
+            .copy(dataRegisteredByDataland = listOf(expectedDataMetaInfo))
+        val getCompaniesByCountryCodeAndSectorResponse = apiAccessor.companyDataControllerApi.getCompanies(
+            apiAccessor.frameworkData,
+            sectors = if (uploadInfo.actualStoredCompany.companyInformation.sector != null) {
+                setOf(uploadInfo.actualStoredCompany.companyInformation.sector!!) } else { null },
+            countryCodes = setOf(uploadInfo.actualStoredCompany.companyInformation.countryCode),
+        )
+        assertTrue(
+            getCompaniesByCountryCodeAndSectorResponse.contains(expectedStoredCompany),
+            "The posted company could not be found in the query results when querying for its country code and sector.",
+        )
+    }
+
+    @Test
+    fun `post a dummy company and check that it is not returned if filtered by a different sector`() {
+        val uploadInfo = apiAccessor.uploadNCompaniesWithoutIdentifiers(1).first()
+        val getCompaniesByCountryCodeAndSectorResponse = apiAccessor.companyDataControllerApi.getCompanies(
+            apiAccessor.frameworkData,
+            sectors = setOf("${uploadInfo.actualStoredCompany.companyInformation.sector}a"),
+            countryCodes = setOf(uploadInfo.actualStoredCompany.companyInformation.countryCode),
+        )
+        assertFalse(
+            getCompaniesByCountryCodeAndSectorResponse.contains(uploadInfo.actualStoredCompany),
+            "The posted company is in the query results," +
+                " even though the country code filter was set to a different country code.",
+
+        )
+    }
+
+    @Test
+    fun `post some dummy companies and check if the number of companies increased accordingly`() {
+        val allCompaniesListSizeBefore = apiAccessor.getNumberOfStoredCompanies()
+        val listOfUploadInfo = apiAccessor.uploadNCompaniesWithoutIdentifiers(3)
+        listOfUploadInfo.forEach {
+            uploadTestDataSet(it.actualStoredCompany.companyId)
+        }
+        val allCompaniesListSizeAfter = apiAccessor.getNumberOfStoredCompanies()
+        assertEquals(
+            listOfUploadInfo.size,
+            allCompaniesListSizeAfter - allCompaniesListSizeBefore,
+            "The size of the all-companies-list did not increase by ${listOfUploadInfo.size}.",
+        )
+    }
+
+    @Test
+    fun `post a dummy company and check if it can be searched for by identifier at the right time`() {
+        val uploadInfo = apiAccessor.uploadOneCompanyWithRandomIdentifier()
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Reader)
+        assertTrue(
+            apiAccessor.companyDataControllerApi.getCompanies(
+                apiAccessor.frameworkData,
```

Comment:
Can be null as soon as dataType is nullable again

Reply:
Done